staging/src/k8s.io/kubectl/pkg/describe/describe.go

```diff
@@ -3049,7 +3052,11 @@ func describeService(service *corev1.Service, endpoints *corev1.Endpoints, event
 			if sp.NodePort != 0 {
 				w.Write(LEVEL_0, "NodePort:\t%s\t%d/%s\n", name, sp.NodePort, sp.Protocol)
 			}
-			w.Write(LEVEL_0, "Endpoints:\t%s\n", formatEndpoints(endpoints, sp.Name))
+			eps := formatEndpointSlices(endpointSlices, sp.Name)
+			if eps == "<none>" {
+				eps = formatEndpoints(endpoints, sp.Name)
+			}
```

Comment:
I don't think this is correct; nothing in kubernetes uses `Endpoints` any more. If there is an Endpoints and no EndpointSlice, then the Service has no endpoints and should be described as such.

(ie, you should drop `formatEndpoints`)

Reply:
one rare corner case is when someone creates a service without selectors, creates endpoints manually and skips mirroring by endpoint slice mirror controller. But I guess can ignore that.
https://github.com/kubernetes/kubernetes/issues/123954#issuecomment-2037497126